### PR TITLE
SAK-45690 Changed INVALID to FAILED. Added transactiontemplates

### PR DESCRIPTION
**Pull request by ottenhoff**

### CODE REVIEW

File: kernel/kernel-impl/src/main/java/org/sakaiproject/content/impl/FileConversionServiceImpl.java [hunk 1]
```diff
@@ -132,12 +140,21 @@ public void startIfEnabled() {
 
                         workers.submit(() -> {
 
-                            log.debug("Setting item with ref {} to IN_PROGRESS ...", ref);
-                            item.setStatus(FileConversionQueueItem.Status.IN_PROGRESS);
-                            item.setAttempts(item.getAttempts() + 1);
-                            item.setLastAttemptStarted(Instant.now());
+                            // This should update the item to IN_PROGRESS and flush the change so
+                            // that other workers can see it. This is the mutex, in effect.
+                            (new TransactionTemplate(transactionManager)).execute(new TransactionCallbackWithoutResult() {
```
Comment (ern) [inline kernel/kernel-impl/src/main/java/org/sakaiproject/content/impl/FileConversionServiceImpl.java:145]: You don't need to to create a template all the time can simply use the existing.

File: kernel/kernel-impl/src/main/java/org/sakaiproject/content/impl/FileConversionServiceImpl.java [hunk 2]
```diff
@@ -159,15 +176,24 @@ public void startIfEnabled() {
...
+                                repository.deleteById(item.getId());
                             } catch (Exception e) {
-                                // Too many attempts. Do not try again.
-                                if (inProgressItem.getAttempts() > maxAttemptsAllowed) {
-                                    inProgressItem.setStatus(FileConversionQueueItem.Status.INVALID);
-                                } else {
-                                    inProgressItem.setStatus(FileConversionQueueItem.Status.NOT_STARTED);
-                                }
-                                repository.save(inProgressItem);
+                                (new TransactionTemplate(transactionManager)).execute(new TransactionCallbackWithoutResult() {
```
Comment (ern) [inline kernel/kernel-impl/src/main/java/org/sakaiproject/content/impl/FileConversionServiceImpl.java:182]: Can actually create the tx template with spring beans.

Comment (ottenhoff) [reply kernel/kernel-impl/src/main/java/org/sakaiproject/content/impl/FileConversionServiceImpl.java:182]: not sure how to do this. @adrianfish ?

**Comment by ern** [comment]
Pushed some changes with respect to TransactionTemplate

**Review by ottenhoff — COMMENTED** [review]
I see. Thanks. Looks good. Important fix

**Review by adrianfish — APPROVED** [review]
GTG

**Pull request merged**